### PR TITLE
feat: support docker login

### PR DESCRIPTION
**Pull request by kaizen3031593**
Adds the ability to login to dockerHub, ecr, or custom repositories.

### CODE REVIEW

File: src/docker-credentials.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
+export class DockerCredential {
+  /**
+   * Create a credential for DockerHub. This method assumes that you will have long-lived
+   * Github Secrets stored under the usernameKey and personalAccessTokenKey.
```
Comment (rix0rrr) [inline src/docker-credentials.ts:4]: Mention the defaults.

File: src/docker-credentials.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+   */
+  public static dockerHub(creds: DockerHubCredentialSecrets = {}): DockerCredential {
+    return new DockerCredential(
+      'docker',
+      undefined,
+      creds.usernameKey ?? 'DOCKERHUB_USERNAME',
```
Comment (rix0rrr) [inline src/docker-credentials.ts:10]: Can you double-check these names are commonly used across the internet?

File: src/docker-credentials.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+      creds.personalAccessTokenKey ?? 'DOCKERHUB_TOKEN',
+    );
+  }
+
+  /**
+   * Create a credential for ECR. This method will reuse your AWS credentials to log in to AWS.
```
Comment (rix0rrr) [inline src/docker-credentials.ts:16]: Reiterate what you mean when you say "your AWS credentials".

File: src/docker-credentials.ts [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+   * @default 'DOCKERHUB_USERNAME'
+   */
+  readonly usernameKey?: string;
+
+  /**
+   * The key of the Github Secret containing the DockerHub personal access token.
+   *
+   * @default 'DOCKERHUB_TOKEN'
+   */
+  readonly personalAccessTokenKey?: string;
```
Comment (rix0rrr) [inline src/docker-credentials.ts:58]: Is this the name DockerHub name?

File: src/docker-credentials.ts (hunk 3 above)
Comment (rix0rrr) [reply src/docker-credentials.ts:16]: Probably also add a description of the IAM permissions this user/role needs.

**Review by rix0rrr — COMMENTED** [review]
Conditionally approved, but I'm not sure that exists on this repo so not approving yet 😝

File: src/docker-credentials.ts (hunk 2 above)
Comment (kaizen3031593) [reply src/docker-credentials.ts:10]: This looks to be the correct defaults. It says so on the official [docker login action](https://github.com/docker/login-action#docker-hub), and comes up often when I search for them on google.

File: src/docker-credentials.ts (hunk 4 above)
Comment (kaizen3031593) [reply src/docker-credentials.ts:58]: we create personal access tokens in DockerHub that is what this value references. I'm asking the user to store the key to the github secret that holds the personal access token, so I think this is an appropriate name.

**Review by rix0rrr — APPROVED** [review]

**Pull request merged**